app/web/src/newhotness/Explore.vue

```diff
@@ -926,10 +949,25 @@ const openAddComponentModal = () => {
 
 const addViewModalRef = ref<InstanceType<typeof AddViewModal>>();
 
-const openViewModal = () => {
+const openAddViewModal = () => {
   addViewModalRef.value?.open();
 };
 
+const editViewModalRef = ref<InstanceType<typeof EditViewModal>>();
+
+const openEditViewModal = (viewId: string) => {
+  // FIXME(nick): right now, we need to switch the current view to edit it because we need the query data.
+  // Why? The current view may differ than the one wishing to be edited. I had a version of this where the
+  // modal had an inner query to account for this scenario (essentially two different "view component list"
+  // calls), but I scrapped it since the query would never load properly. Ideally, you should be able to
+  // delete a view or edit it without switching to it. For now, we will survive.
+  selectedView.value = viewId;
```

Comment:
I did not want to do this, but spinning the wheels here isn't worth it at present. It would be nice to have a better way to have conditional query loading like "you asked for a different view than the one loaded in the Explore Vue component, okay I'll get its components", but doing that is a bit jank right now... for now, let's just switch to the view so that we get all of its components needed for the "canDeleteView" computation.